src/index/TextIndexBuilder.cpp

```diff
@@ -235,96 +239,33 @@ void TextIndexBuilder::logEntityNotFound(const std::string& word,
+                                       EntityTextVec& entityTextVec) {
   ad_utility::File out(filename.c_str(), "w");
-  currenttOffset_ = 0;
-  // Detect block boundaries from the main key of the vec.
-  // Write the data for each block.
-  // First, there's the classic lists, then the additional entity ones.
-  TextBlockIndex currentBlockIndex = 0;
-  WordIndex currentMinWordIndex = std::numeric_limits<WordIndex>::max();
-  WordIndex currentMaxWordIndex = std::numeric_limits<WordIndex>::min();
-  std::vector<Posting> classicPostings;
-  std::vector<Posting> entityPostings;
-  for (const auto& value : vec.sortedView()) {
-    TextBlockIndex textBlockIndex = value[0].getInt();
-    bool flag = value[1].getBool();
-    TextRecordIndex textRecordIndex = TextRecordIndex::make(value[2].getInt());
-    WordOrEntityIndex wordOrEntityIndex = value[3].getInt();
-    Score score = static_cast<Score>(value[4].getDouble());
-    if (textBlockIndex != currentBlockIndex) {
-      AD_CONTRACT_CHECK(!classicPostings.empty());
-      bool scoreIsInt = textScoringMetric_ == TextScoringMetric::EXPLICIT;
-      ContextListMetaData classic = textIndexReadWrite::writePostings(
-          out, classicPostings, currenttOffset_, scoreIsInt);
-      ContextListMetaData entity = textIndexReadWrite::writePostings(
-          out, entityPostings, currenttOffset_, scoreIsInt);
-      textMeta_.addBlock(TextBlockMetaData(
-          currentMinWordIndex, currentMaxWordIndex, classic, entity));
-      classicPostings.clear();
-      entityPostings.clear();
-      currentBlockIndex = textBlockIndex;
-      currentMinWordIndex = wordOrEntityIndex;
-      currentMaxWordIndex = wordOrEntityIndex;
-    }
-    if (!flag) {
-      classicPostings.emplace_back(textRecordIndex, wordOrEntityIndex, score);
-      if (wordOrEntityIndex < currentMinWordIndex) {
-        currentMinWordIndex = wordOrEntityIndex;
-      }
-      if (wordOrEntityIndex > currentMaxWordIndex) {
-        currentMaxWordIndex = wordOrEntityIndex;
-      }
-
-    } else {
-      entityPostings.emplace_back(textRecordIndex, wordOrEntityIndex, score);
-    }
-  }
-  // Write the last block
-  AD_CONTRACT_CHECK(!classicPostings.empty());
-  bool scoreIsInt = textScoringMetric_ == TextScoringMetric::EXPLICIT;
-  ContextListMetaData classic = textIndexReadWrite::writePostings(
-      out, classicPostings, currenttOffset_, scoreIsInt);
-  ContextListMetaData entity = textIndexReadWrite::writePostings(
-      out, entityPostings, currenttOffset_, scoreIsInt);
-  textMeta_.addBlock(TextBlockMetaData(currentMinWordIndex, currentMaxWordIndex,
-                                       classic, entity));
-  classicPostings.clear();
-  entityPostings.clear();
+  auto textBlockWriter = TextBlockWriter{wordTextVec, entityTextVec, out,
+                                         textScoringMetric_, textMeta_};
+  textBlockWriter.calculateAndWriteTextBlocks(nofWordPostingsInTextBlock_);
```

Comment:
This can be a static function `TextBlockWriter::writeTextIndex(filename, tec1, tex2, metric, metadata)`, 

Then you save some additional lines in this class.